### PR TITLE
Add yum provider support

### PR DESCRIPTION
**Pull request by drexin**

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@aciidb0mb3r any idea why this is failing?

```
/home/buildnode/jenkins/workspace/swift-package-manager-self-hosted-Linux-smoke-test/branch-master/swiftpm/Sources/PackageLoading/Target+PkgConfig.swift:127:39: error: type 'Platform.LinuxFlavor' has no member 'fedora'
16:53:02             if case .linux(.fedora) = platform {
16:53:02                                       ^~~~~~~~
```

The case is clearly part of the PR: https://github.com/apple/swift-package-manager/pull/2642/files#diff-72f5fe00a512bdefecfdb4d0a0574387R23

**Comment by aciidb0mb3r** [comment]
Ah, you need to land the TSC changes in this repository first: https://github.com/apple/swift-tools-support-core

File: Sources/PackageDescription/PackageDescription.swift [hunk 1]
```diff
@@ -456,6 +458,19 @@ public enum SystemPackageProvider {
         return ._aptItem(packages)
       #endif
     }
+
+    /// Declare the list of installable packages using the yum package
+    /// manager on RHEL/CentOS to create a system package provider instance.
+    ///
+    /// - Parameters:
+    ///     - packages: The list of package names.
+    public static func yum(_ packages: [String]) -> SystemPackageProvider {
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription/PackageDescription.swift]: This should be marked using package description availability: `@available(_PackageDescription, introduced: 999)`

**Comment by drexin** [comment]
Ah, good. Thanks!

**Comment by drexin** [comment]
@aciidb0mb3r Created https://github.com/apple/swift-tools-support-core/pull/59

Should I remove the code from this PR?

**Comment by aciidb0mb3r** [comment]
> Should I remove the code from this PR?

Nah, it needs to be in both places right now :(

File: Sources/PackageDescription/PackageDescription.swift [hunk 2]
```diff
@@ -426,9 +426,11 @@ public enum SystemPackageProvider {
   #if PACKAGE_DESCRIPTION_4
     case brewItem([String])
     case aptItem([String])
+    case yumItem([String])
```
Comment (aciidb0mb3r) [inline Sources/PackageDescription/PackageDescription.swift]: We shouldn't add this case for PackageDescription 4.

File: Tests/BuildTests/BuildPlanTests.swift [hunk 3]
```diff
@@ -1412,6 +1412,7 @@ final class BuildPlanTests: XCTestCase {
                             providers: [
                                 .brew(["BTarget"]),
                                 .apt(["BTarget"]),
+                                .yum(["BTarget"]),
                             ]
                         )
                     ]),
```
Comment (aciidb0mb3r) [inline Tests/BuildTests/BuildPlanTests.swift]: Hmm, not sure what's up with the commented out assertion. Can you check?

**Comment by aciidb0mb3r** [comment]
Some minor comments but LGTM! Thanks for working on this 😃

File: Sources/PackageDescription/PackageDescription.swift (hunk 2 above)
Comment (drexin) [reply Sources/PackageDescription/PackageDescription.swift]: ok

File: Tests/BuildTests/BuildPlanTests.swift (hunk 3 above)
Comment (drexin) [reply Tests/BuildTests/BuildPlanTests.swift]: I'll open a separate PR for that, as it's unrelated to this change. If for some reason this has to be reverted, we would regress there again.

**Comment by drexin** [comment]
@aciidb0mb3r thanks for the review. I removed the PD4 cases.

**Comment by drexin** [comment]
@swift-ci test

**Comment by drexin** [comment]
@swift-ci smoke test

**Pull request merged**